azure-identity/azure/identity/credentials/providers.py

```diff
@@ -0,0 +1,185 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from collections import namedtuple
+import os
+from time import time
+from typing import Any, Iterable, Mapping, Optional, TYPE_CHECKING  # pylint:disable=unused-import
+
+from azure.core import Configuration, HttpRequest
+from azure.core.pipeline import Pipeline
+from azure.core.pipeline.policies import ContentDecodePolicy, HeadersPolicy, NetworkTraceLoggingPolicy, RetryPolicy
+from azure.core.pipeline.transport import RequestsTransport
+from msal import TokenCache
+
+from .exceptions import AuthenticationError
+
+if TYPE_CHECKING:
+    # pylint:disable=unused-import
+    from typing import Iterable, Optional, Tuple, Union
+    from azure.core import Configuration
+    from ._protocols import SupportsGetToken
+
+# pylint:disable=too-few-public-methods
+
+
+class ClientSecretTokenProvider:
```

Comment:
I would really like to standardize on calling these XxxCredential across all languages. In this case ClientSecretCredential